src/get_climate_future.R

```diff
@@ -0,0 +1,184 @@
+            pivot_wider(names_from = "var", 
+                        values_from = "val") %>%
+            dplyr::mutate(date = as.Date(date),
+                   ppt_mm = pr,
+                   tmax_C = tasmax - 273.15,
+                   tmin_C = tasmin - 273.15,
+                   tmean_C = (tmax_C + tmin_C)/2,
+                   join_index = aoi$join_index) %>%
+            dplyr::select(-c("tasmax","tasmin","pr"))
+    
+    } else {
+      
+      all_climate_data[[idx]] <- clim %>%
+        data.table() %>%
+        # since polygon grabbed a single grid, does not provide the coordinates
+        # of the gridMET cell, so we fill in x and y with the coordinates
+        # of the sf object:
+        dplyr::mutate(x = sf::st_coordinates(aoi)[[1]],
+                      y = sf::st_coordinates(aoi)[[2]]) %>%
+        # Then do all other cleaning steps done for polygon sf objects:
+        dplyr::mutate(date = as.Date(date),
+                      ppt_mm = pr,
+                      tmax_C = tasmax - 273.15,
+                      tmin_C = tasmin - 273.15,
+                      tmean_C = (tmax_C + tmin_C)/2,
+                      join_index = aoi$join_index) %>%
+        dplyr::select(-c("tasmax","tasmin","pr"))
+      
+     }
+    }
+  }
+  
+  all_climate_data <- all_climate_data %>%
+    bind_rows()
+  
+  # Rename the join_index column
+  colnames(all_climate_data)[colnames(all_climate_data) == "join_index"] <- {{col_name}}
+  
+  return(all_climate_data)
+  
+  
+  # IF the input geometry is a point...
+} else if(unique(sf::st_geometry_type(sf)) == "POINT") {
+  
+    for (i in 1:nrow(sf)) {
+      
+      aoi <- sf[i,]
+      
+      for (j in 1:length(GCM)) {
+        
+        idx <- idx + 1
+        
+        print(paste0('Downloading MACA for ', aoi$join_index, "."))
+        
+        clim <- climateR::getMACA(AOI = aoi, 
+                                  varname = vars,#,"pet","vs"),
+                                  model = models[j],
+                                  scenario = scenarios[j],
+                                  startDate = start,
+                                  endDate = end) 
+    
+        all_climate_data[[idx]] <- clim %>%
+          data.table() %>%
+          # since point pulls from gridMET do not provide the coordinates
+          # of the gridMET cell, we fill in x and y with the coordinates
+          # of the sf object:
+          dplyr::mutate(x = sf::st_coordinates(aoi)[[1]],
+                        y = sf::st_coordinates(aoi)[[2]]) %>%
+          # Then do all other cleaning steps done for polygon sf objects:
+          dplyr::mutate(date = as.Date(date),
+                        ppt_mm = pr,
+                        tmax_C = tasmax - 273.15,
+                        tmin_C = tasmin - 273.15,
+                        tmean_C = (tmax_C + tmin_C)/2,
+                        join_index = aoi$join_index) %>%
+          dplyr::select(-c("tasmax","tasmin","pr"))
```

Comment:
all_climate_data[[i]] <- clim %>%
          data.table() %>%
          # names of columns include va_mode_rcp so must rename
          rename_with(~ str_split(.x, "_", n = 2) %>% map_chr(1)) %>%
          # since point pulls from gridMET do not provide the coordinates
          # of the gridMET cell, we fill in x and y with the coordinates
          # of the sf object:
          dplyr::mutate(x = sf::st_coordinates(aoi)[[1]],
                        y = sf::st_coordinates(aoi)[[2]]) %>%
          # Then do all other cleaning steps done for polygon sf objects:
          dplyr::mutate(date = as.Date(date),
                        ppt_mm = pr,
                        tmax_C = tasmax - 273.15,
                        tmin_C = tasmin - 273.15,
                        tmean_C = (tmax_C + tmin_C)/2,
                        join_index = aoi$join_index,
                        GCM = models[j],
                        RCP = scenarios[j],
                        Ensemble = "r1i1p1") %>%
          dplyr::select(-c("tasmax","tasmin","pr"))